File: Kotlin/002_Program/Main.kt
fun main() {
    println("=== Kotlin Calculator ===")
    print("Enter first number: ")
    val num1 = readLine()?.toDoubleOrNull() ?: 0.0

    print("Enter operator (+, -, *, /): ")
    val operator = readLine() ?: "+"

    print("Enter second number: ")
    val num2 = readLine()?.toDoubleOrNull() ?: 0.0

    val result = when (operator) {
        "+" -> num1 + num2
        "-" -> num1 - num2
        "*" -> num1 * num2
        "/" -> if (num2 != 0.0) num1 / num2 else Double.NaN
        else -> Double.NaN
    }

    println("Result: $num1 $operator $num2 = $result")
}

File: Swift/002_Program/main.swift
import Foundation

print("=== Swift Calculator ===")
print("Enter first number: ", terminator: "")
let num1 = Double(readLine() ?? "0") ?? 0.0

print("Enter operator (+, -, *, /): ", terminator: "")
let op = readLine() ?? "+"

print("Enter second number: ", terminator: "")
let num2 = Double(readLine() ?? "0") ?? 0.0

var result: Double = 0.0

switch op {
case "+":
    result = num1 + num2
case "-":
    result = num1 - num2
case "*":
    result = num1 * num2
case "/":
    result = num2 != 0 ? num1 / num2 : 0
default:
    print("Invalid operator")
}

print("Result: \(num1) \(op) \(num2) = \(result)")
